docker-compose.yml

```diff
@@ -75,6 +75,35 @@ services:
     volumes:
       - ./tests/mongo:/tests/mongo
 
+  # used for testing a replica set
+  mongodb_replset:
+    build:
+      context: ./build/deps
+      dockerfile: ${MONGO_DOCKERFILE:-mongo6}.Dockerfile
+    container_name: mongodb_replset
+    command: mongod --port 47017 --bind_ip_all --replSet rs0
+    ports:
+      - 47017:27017
+    extra_hosts:
+    - "host.docker.internal:host-gateway"
+    environment:
+      # Always UTC+05:45. Set to catch timezone problems.
+      - TZ=Asia/Kathmandu
+    volumes:
+      - ./build/mongod_repl.conf:/etc/mongod.conf
+  
+  # initiates the replica set
+  # we need a separate container to run rs.initiate()
+  init_mongodb_replset:
+    build:
+      context: ./build/deps
+      dockerfile: ${MONGO_DOCKERFILE:-mongo6}.Dockerfile
+    container_name: init_mongodb_replset
+    depends_on:
+      - mongodb_replset
+    restart: "no"
+    entrypoint: ["bash", "-c", "sleep 10 && mongosh --host mongodb_replset:47017 --eval 'rs.initiate();'"]
```

Comment:
I see you do `sleep 10` to ensure that `mongodb_replset` is ready. In principle, a better approach would be to configure [startup order](https://docs.docker.com/compose/startup-order/) with some sort of readiness check. However, I don't have a quick way to add such a readiness check to this config, so if `sleep 10` is enough, we can keep it for now.